### PR TITLE
Upload projects to hale connect

### PR DESCRIPTION
**Pull request by florianesser**
Support for uploading transformation projects to hale connect, as outlined in #292.

### CODE REVIEW

**Comment by florianesser** [comment]
Also contains commits from #355 at the moment. Will rebase after #355 is completed and merged.

File: io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/HaleConnectService.java [hunk 1]
```diff
@@ -138,4 +185,70 @@
...
+	 * @param projectId Transformation project ID
+	 * @param owner Project owner
+	 * @param file the file to upload
+	 * @param progress a progress indicator
+	 * @return a {@link ListenableFuture} whose value will be set to true upon
+	 *         successful completion of the request or false when the request
+	 *         fails
+	 * @throws HaleConnectException thrown on any API exception
+	 */
+	ListenableFuture<Boolean> uploadProjectFileAync(String projectId, Owner owner, File file,
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/HaleConnectService.java]: +s?

Comment (florianesser) [inline io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/HaleConnectService.java]: The method actually uploads only a single file (which may be an archive though).

Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/HaleConnectService.java]: Ah, sorry, I meant the **A`s`ync** part.

**Pull request merged**